statsmodels/discrete/discrete_model.py

```diff
@@ -1173,6 +1174,280 @@ def hessian(self, params):
+        self._transparams = False
+        if start_params is None:
+            offset = getattr(self, "offset", 0) + getattr(self, "exposure", 0)
+            if np.size(offset) == 1 and offset == 0:
+                offset = None
+            mod_poi = Poisson(self.endog, self.exog, offset=offset)
+            start_params = mod_poi.fit_regularized(
+                start_params=start_params, method=method, maxiter=maxiter,
+                full_output=full_output, disp=0, callback=callback,
+                alpha=alpha_p, trim_mode=trim_mode, auto_trim_tol=auto_trim_tol,
+                size_trim_tol=size_trim_tol, qc_tol=qc_tol, **kwargs).params
+            start_params = np.append(start_params, 1.2)
+
+        cntfit = super(CountModel, self).fit_regularized(
+                start_params=start_params, method=method, maxiter=maxiter,
+                full_output=full_output, disp=disp, callback=callback,
+                alpha=alpha, trim_mode=trim_mode, auto_trim_tol=auto_trim_tol,
+                size_trim_tol=size_trim_tol, qc_tol=qc_tol, **kwargs)
+
+        if method in ['l1', 'l1_cvxopt_cp']:
+            discretefit = L1GeneralizedPoissonResults(self, cntfit)
+        else:
+            raise Exception(
+                    "argument method == %s, which is not handled" % method)
+
+        return L1GeneralizedPoissonResultsWrapper(discretefit)
+
+    fit_regularized.__doc__ = DiscreteModel.fit_regularized.__doc__
+
+    def score(self, params):
+        """
+        Generalized Poisson model score (gradient) vector of the log-likelihood
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model
+
+        Returns
+        -------
+        score : ndarray, 1-D
+            The score vector of the model, i.e. the first derivative of the
+            loglikelihood function, evaluated at `params`
+
+        Notes
+        -----
+        """
+        return approx_fprime(params, self.loglikeobs).sum(0)
+        
+        if self._transparams: # got lnalpha during fit
+            alpha = np.exp(params[-1])
+        else:
+            alpha = params[-1]
+        params = params[:-1]
+        p = self.parameterization
+        exog = self.exog
+        endog = self.endog[:,None]
+        mu = self.predict(params)[:,None]
+        mu_p = np.power(mu, p)
+        mu_y = mu_p * endog
+        alphamu = 1 + alpha * mu_p
+        alphamu_y = mu + (alphamu - 1) * endog
+        dalpha = (mu_y * ((endog - 1) / alphamu_y - 1 / alphamu) - 
+                  mu_y / alphamu - alphamu_y * mu_p / (alphamu **2 ))
+        palphamu = alpha * p * mu ** (p - 1)
+        palphamu_y = palphamu * endog
+        dparams = exog * (1 / mu + (endog - 1) * (1 + palphamu_y) / alphamu_y -
+                   (endog * palphamu / alphamu) -
+                   (1 + palphamu_y) / alphamu + alphamu_y * palphamu / 
+                   (alphamu * alphamu))
+        return np.r_[dparams.sum(0), dalpha.sum()]
```

Comment:
If you didn't sum() before returning, this could be a `score_obs`.  Doesn't make that big a difference, but its nice to have the flexibility.  (Same is true for NegativeBinomial I think)